test/UnionGeneration/FactoryMethodTests.cs

```diff
@@ -0,0 +1,113 @@
+namespace Dunet.Test.UnionGeneration;
+
+public class FactoryMethodTests
+{
+    [Fact]
+    public void NonGenericFactoryMethods()
+    {
+        var programCs = """
+            using Dunet;
+            using System;
+
+            Result success = Result.AsSuccess("Hello, world!");
```

Comment:
```suggestion
            Result success = Result.OfSuccess("Hello, world!");
```
What do you think about this naming instead? Using `As` implies some kind of casting or conversion occurring like when we use the `as` keyword.

Reply:
Good point, I'm happy with `OfX`. I'll update it later today unless you want to push it to this branch - either I'm fine with :+1:

Reply:
Go for it. I'll approve once the methods have been renamed 👍

Reply:
Done in 71fb5ee